game-core/src/main/java/games/strategy/engine/message/RemoteMethodCall.java

```diff
@@ -94,6 +95,12 @@ public String getMethodName() {
   }
 
   private static String[] classesToString(final Class<?>[] classes, final Object[] args) {
+    if (args != null && classes.length != args.length) {
```

Comment:
@DanVanAtta  This is the precondition I meant, it should prevent the `ArrayIndexOutOfBoundsException` reported in https://github.com/triplea-game/triplea/issues/6104#issuecomment-610713513